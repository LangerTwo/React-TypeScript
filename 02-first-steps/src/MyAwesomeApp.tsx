import type { CSSProperties } from "react";

const firstName = 'Arthur';
const lastName = 'Ainsworth';

const favotiteGames = ['Metal Gear', 'Death Stranding', 'Elden Ring'];
const isActive = true;

const address = {
    zipCode: 'ABC-123',
    country: 'Canad√°',
};
const myStyle: CSSProperties = {
    backgroundColor: '#fafafa',
    borderRadius: 10,
    padding: 10,
    marginTop: 20,
};
export function MyAwesomeApp() {


    return (
        <>
            <h1 data-testid="first-name-tittle">{ firstName }</h1>
            <h3>{ lastName }</h3>

            <p className="mi-clase-favorita">{favotiteGames.join(', ')}</p>
            <p>{ 2 + 2 }</p>

            <h1>{isActive ? 'Activo' : 'No activo'}</h1>

            <p
                style={myStyle}
            >{JSON.stringify(address)}</p>
        </>
    );
};