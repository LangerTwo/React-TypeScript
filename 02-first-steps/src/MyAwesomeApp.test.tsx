import { describe, expect, test } from "vitest";
import { render, screen } from "@testing-library/react";


import { MyAwesomeApp } from "./MyAwesomeApp";


describe('MyAwesomeApp', () => {
    test('should render firsName and lastName', () => {
        const { container } = render(<MyAwesomeApp />)
        // screen.debug();
        // console.log(container.innerHTML);

        const h1 = container.querySelector('h1');
        const h3 = container.querySelector('h3');

        expect(h1?.innerHTML).toContain('Arthur');
        expect(h3?.innerHTML).toContain('Ainsworth');
    });

    test('should render firsName and lastName - screen', () => {
        render(<MyAwesomeApp />);
        screen.debug()
        // console.log(container.innerHTML);

        // const h1 = screen.getByRole('heading', {
        //     level: 1
        // });

        const h1 = screen.getByTestId('first-name-tittle');

        expect(h1?.innerHTML).toContain('Arthur');
    });

    test('should match snapshot', () => {
        const { container } = render(<MyAwesomeApp />);

        expect(container).toMatchSnapshot();
    })
});