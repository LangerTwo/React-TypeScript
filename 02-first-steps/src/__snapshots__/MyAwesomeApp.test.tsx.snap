// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <h1
    data-testid="first-name-tittle"
  >
    Arthur
  </h1>
  <h3>
    Ainsworth
  </h3>
  <p
    class="mi-clase-favorita"
  >
    Metal Gear, Death Stranding, Elden Ring
  </p>
  <p>
    4
  </p>
  <h1>
    Activo
  </h1>
  <p
    style="background-color: rgb(250, 250, 250); border-radius: 10px; padding: 10px; margin-top: 20px;"
  >
    {"zipCode":"ABC-123","country":"Canad√°"}
  </p>
</div>
`;
